docs/lang_spec/Cadence Language Specification.md

```diff
@@ -0,0 +1,5798 @@
+
+The primary goals of this specification are to:
+- Provide an unambiguous description of Cadence's structure and behavior.
+- Facilitate the development of tooling, such as compilers, interpreters, and debuggers, that conform to the same standards.
+- Support developers in writing secure, efficient, and maintainable smart contracts.
+- Guide the evolution of Cadence by ensuring changes and new features maintain compatibility and correctness within the language framework.
+
+The intended audience of this specification includes:
+- **Language Implementers**: Individuals or teams responsible for creating or maintaining Cadence compilers, interpreters, and other development tools.
+- **Developers**: Software engineers writing smart contracts or dApps using Cadence on the Flow blockchain.
+- **Tool Makers**: Developers building integrated development environments (IDEs), testing tools, and utilities that work with Cadence.
+
+### 2. Overview of Cadence
+Cadence is a resource-oriented programming language that prioritizes safety, clarity, and developer experience in the context of blockchain development. Designed by Flow Foundation, Cadence enables secure and scalable development of smart contracts, particularly for applications involving digital assets such as non-fungible tokens (NFTs) and decentralized finance (DeFi) protocols.
+
+The primary design philosophy behind Cadence is **resource-oriented programming**, which introduces the concept of resources as first-class citizens. Resources represent digital assets that are protected by strict rules ensuring that they cannot be accidentally duplicated or lost. By enforcing ownership and access control at the language level, Cadence minimizes many common security vulnerabilities found in blockchain programming.
+
+Cadence is the core language for smart contracts on the Flow blockchain, a platform built to support high-performance dApps for mainstream adoption. Flow's account-based model, combined with Cadence's strong typing and resource management features, provides a secure and user-friendly environment for developers to create complex decentralized applications.
+
+In comparison to other blockchain languages like Solidity (used on Ethereum), Cadence emphasizes **safety** and **developer ergonomics**. While Solidity relies on more traditional programming paradigms, Cadence’s resource-oriented model provides built-in safeguards against common issues like unintended re-entrancy or resource mismanagement, making it especially suitable for managing valuable digital assets.
```

Comment:
Great idea to point out the design philosophy and goals.

The context of Cadence having been designed and built as part of Flow, as well as comparisons to other languages, are not really relevant in the language specification, maybe rephrase and remove:

```suggestion
Cadence is a resource-oriented programming language that prioritizes safety, clarity, and developer experience in the context of blockchain development. Cadence aims to allow for secure and scalable development of smart contracts, particularly for applications involving digital assets such as non-fungible tokens (NFTs) and decentralized finance (DeFi) protocols.

The primary design philosophy behind Cadence is **resource-oriented programming**, which introduces the concept of resources as first-class citizens. Resources represent digital assets that are protected by strict rules ensuring that they cannot be accidentally duplicated or lost. By enforcing ownership and access control at the language level, Cadence minimizes many common security vulnerabilities found in blockchain programming.
```